wdl-analysis/src/analyzer.rs

```diff
@@ -285,28 +329,30 @@ impl SourceEdit {
                 .offset(end)
                 .expect("expected a valid end position")
                 .into();
-        source.replace_range(range, &self.text);
+
+        if source.is_char_boundary(range.start) && source.is_char_boundary(range.end) {
+            source.replace_range(range, &self.text);
+        }
```

Comment:
If we decide not to return a `Result` here, then failure to apply should at least be
logged somewhere so it can eventually be uncovered.